### PR TITLE
ML CI: update image

### PR DESCRIPTION
**Pull request by adamjstewart**
Trying to see if the image I created in https://github.com/spack/gitlab-runners/pull/50 is sufficient for our ML pipelines.

### CODE REVIEW

**Review by kwryankrattiger — COMMENTED** [review]
LGTM after merging https://github.com/spack/gitlab-runners/pull/50

File: share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml
Comment (kwryankrattiger) [inline share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: I think this should have a date tag on it right?

Comment (adamjstewart) [reply share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: We haven't created any date tags yet. Is there any reason not to just use main since we control the pipeline and the image?

Comment (kwryankrattiger) [reply share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: my main concerns are just tracking and avoiding accidental upgrades. One of the worst things to have to debug in CI is an untracked container update that makes a minor breaking change...If we can get a date tag that would be good.

Comment (adamjstewart) [reply share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: @haampie @alalazo does one of you have permission to create a new release/tag on https://github.com/spack/gitlab-runners?

Comment (alalazo) [reply share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: I do. What do we need, just a tag of the current main branch?

Comment (alalazo) [reply share/spack/gitlab/cloud_pipelines/.gitlab-ci.yml]: @adamjstewart https://github.com/spack/gitlab-runners/actions/runs/8988272315

**Review by kwryankrattiger — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
